server/src/services/change_file_content.rs

```diff
@@ -9,6 +9,9 @@ pub async fn handle(
     server_state: &mut ServerState,
     request: ChangeFileContentRequest,
 ) -> Result<ChangeFileContentResponse, ChangeFileContentError> {
+    if !request.username.chars().all(|x| x.is_digit(36)) {
```

Comment:
You just need to make sure no one can insert an invalid username, then you don't need to check it ever again. If the username is not in the db, then it's invalid. If it's in the db it must have been valid.